Tests/Services/TasksServiceTests.cs

```diff
@@ -415,4 +408,412 @@ await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+                                                                                                   1,
+                                                                                                   2,
+                                                                                                   3,
+                                                                                                 })
+                                                    },
+                                                  },
+                      },
+                    };
+
+    var result = await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+                                                    taskNodes);
+
+    ClassicAssert.IsNotNull(result);
+    ClassicAssert.AreEqual("taskId1",
+                           result.First()
+                                 .TaskId);
+    ClassicAssert.AreEqual("payloadId1",
+                           result.First()
+                                 .PayloadId);
+    ClassicAssert.AreEqual("outputId1",
+                           result.First()
+                                 .ExpectedOutputs.First());
+  }
+
+  [Test]
+  public void GetTasksDetailedAsyncWithNonExistentTaskIdThrowsException()
+  {
+    var mockInvoker = new Mock<CallInvoker>();
+    mockInvoker.Setup(invoker => invoker.AsyncUnaryCall(It.IsAny<Method<GetTaskRequest, GetTaskResponse>>(),
+                                                        It.IsAny<string>(),
+                                                        It.IsAny<CallOptions>(),
+                                                        It.IsAny<GetTaskRequest>()))
+               .Throws(new RpcException(new Status(StatusCode.NotFound,
+                                                   "Task not found")));
+
+    var taskService = MockHelper.GetTasksServiceMock(mockInvoker);
+
+    Assert.ThrowsAsync<RpcException>(() => taskService.GetTasksDetailedAsync("nonExistentTaskId"));
+  }
+
+  [Test]
+  public void CancelTasksAsyncWithNonExistentTaskIdsThrowsException()
+  {
+    var mockInvoker = new Mock<CallInvoker>();
+    mockInvoker.Setup(invoker => invoker.AsyncUnaryCall(It.IsAny<Method<CancelTasksRequest, CancelTasksResponse>>(),
+                                                        It.IsAny<string>(),
+                                                        It.IsAny<CallOptions>(),
+                                                        It.IsAny<CancelTasksRequest>()))
+               .Throws(new RpcException(new Status(StatusCode.NotFound,
+                                                   "Task not found")));
+
+    var taskService = MockHelper.GetTasksServiceMock(mockInvoker);
+
+    var taskIds = new List<string>
+                  {
+                    "nonExistentTaskId1",
+                    "nonExistentTaskId2",
+                  };
+
+    Assert.ThrowsAsync<RpcException>(() => taskService.CancelTasksAsync(taskIds));
```

Comment:
Please use the [Contraint model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html) of assertions.